### PR TITLE
Table tests & more refactor

### PR DESCRIPTION
**Pull request by caffodian**
Botched some refactoring in sections that weren't actually tested in the previous PR.  Add table related tests before realizing the list fix might have fixed the table cases.

### CODE REVIEW

**Comment by caffodian** [comment]
I thought this was a different class of problem than the lists, but apparently it was the same.  Possibly because `td` are similar to `li` in that they can contain `p` or bare text nodes.

File: plugins/nanospell/plugin.js [hunk 1]
```diff
@@ -366,7 +366,7 @@
 					editor.getSelection().selectBookmarks(bookmarks);
 				}
 
-				triggerSpelling((spell_fast_after_spacebar && (ch8r === 32 || ch8r === 10 || ch8r === 13)))
+				triggerSpelling((spellFastAfterSpacebar && (ch8r === 32 || ch8r === 10 || ch8r === 13)))
```
Comment (winhamwr) [inline plugins/nanospell/plugin.js]: Could we make this magic numbers into constants? I don't know what they mean.

File: plugins/nanospell/plugin.js [hunk 2]
```diff
@@ -742,6 +726,21 @@
...
+				'span',
+				{
+					attributes: {
+						'class': 'nanospell-typo'
+					}
+				}
+			);
+
+			range.extractContents().appendTo(span);
+
```
Comment (winhamwr) [inline plugins/nanospell/plugin.js]: Can we remove the random blank lines between and after these?

File: tests/typowalking/markAllTyposTables.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	},
+	tearDown: function() {
+		this.editorBot.editor.plugins.nanospell.markTypos.restore();
+	},
+	'test marking a simple table': function() {
+		var bot = this.editorBot,
+			blocksToBeMarked,
+			markedHtml;
+
+		bot.setHtmlWithSelection(
```
Comment (winhamwr) [inline tests/typowalking/markAllTyposTables.js:48]: Hooray indentation!

**Comment by winhamwr** [comment]
Nice. It's good to see that the list fix also fixed tables. That's confidence-inducing :)

**Pull request merged**